nuxeo-features/nuxeo-automation/nuxeo-automation-scripting/src/main/java/org/nuxeo/automation/scripting/internals/AutomationScriptingParamsInjector.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.nuxeo.automation.scripting.internals;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationDocumentation.Param;
+
+public interface AutomationScriptingParamsInjector {
+    public void inject(Map<String,Object> params, OperationContext ctx, ScriptingOperationDescriptor desc);
+
+    static AutomationScriptingParamsInjector newInstance(boolean inlinedContext) {
+        if (inlinedContext) {
+            return new AutomationScriptingParamsInjector() {
+                @Override
+                public void inject(Map<String,Object> parms, OperationContext ctx, ScriptingOperationDescriptor desc) {
+                    parms.putAll(ctx);
+                }
+            };
+        }
+        return new AutomationScriptingParamsInjector() {
+
+            @Override
+            public void inject(Map<String,Object> parms, OperationContext ctx, ScriptingOperationDescriptor desc) {
+                Arrays.stream(desc.getParams()).map(Param::getName).filter(name -> ctx.containsKey(name))
+                        .forEach(name -> parms.put(name, ctx.get(name)));
+            }
+        };
```

Comment:
Simplify to:

        if (inlinedContext) {
            return (params, ctx, desc) -> params.putAll(ctx);
        } else {
            return (params, ctx, desc) -> //
            Arrays.stream(desc.getParams()).map(Param::getName).filter(ctx::containsKey).forEach(
                    name -> params.put(name, ctx.get(name)));
        }